lib/SharedLib/CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+# Runtime libs from tt-mlir
+set(TTNN_RUNTIME_LIBS TTRuntime TTRuntimeTTNN)
+
+# Dependency libs from tt-metal/ttnn project for ttnn runtime
+set(TTNN_LIBS TTMETAL_LIBRARY DEVICE_LIBRARY TTEAGER_LIBRARY TTNN_LIBRARY)
+
+# Libs from tt-mlir project
+set(TTMLIR_LIBS
+    TTNNTargetFlatbuffer
+    MLIRTTDialect
+    MLIRTTIRDialect
+    MLIRTTNNDialect
+    MLIRTTKernelDialect
+    TTMLIRTTIRToTTNN
+    MLIRTTMetalDialect
+    MLIRTTIRTransforms
+    MLIRTTNNTransforms
+    MLIRTTIRAnalysis
+    MLIRTTNNPipelines
+    TTMLIRTTNNToEmitC
+)
+
+# We supply empty.cpp because CMake does not allow creating a library without sources.
+add_library(TTMLIR SHARED empty.cpp)
```

Comment:
I think you can work around this issue of must having sources using interface lib https://cmake.org/cmake/help/latest/command/add_library.html#interface-libraries.